src/pytorch_lightning/callbacks/lr_finder.py

```diff
@@ -0,0 +1,99 @@
+r"""
+LearningRateFinder
+==================
+
+Finds optimal learning rate
+"""
+from typing import Optional
+
+import pytorch_lightning as pl
+from pytorch_lightning.callbacks.callback import Callback
+from pytorch_lightning.tuner.lr_finder import _LRFinder, lr_find
+from pytorch_lightning.utilities.exceptions import _TunerExitException, MisconfigurationException
+from pytorch_lightning.utilities.seed import isolate_rng
+
+
+class LearningRateFinder(Callback):
+    """The ``LearningRateFinder`` callback enables the user to do a range test of good initial learning rates, to
+    reduce the amount of guesswork in picking a good starting learning rate.
+
+    Args:
+        min_lr: Minimum learning rate to investigate
+
+        max_lr: Maximum learning rate to investigate
+
+        num_training: Number of learning rates to test
+
+        mode: Search strategy to update learning rate after each batch:
+
+            - ``'exponential'`` (default): Increases the learning rate exponentially.
+            - ``'linear'``: Increases the learning rate linearly.
+
+        early_stop_threshold: Threshold for stopping the search. If the
+            loss at any point is larger than early_stop_threshold*best_loss
+            then the search is stopped. To disable, set to None.
+
+        update_attr: Whether to update the learning rate attribute or not.
+
+    Raises:
+        MisconfigurationException:
+            If learning rate/lr in ``model`` or ``model.hparams`` isn't overridden when ``auto_lr_find=True``,
+            or if you are using more than one optimizer.
+    """
+
+    SUPPORTED_MODES = ("linear", "exponential")
+
+    def __init__(
+        self,
+        min_lr: float = 1e-8,
+        max_lr: float = 1,
+        num_training_batches: int = 100,
+        mode: str = "exponential",
+        early_stop_threshold: float = 4.0,
+        update_attr: bool = False,
+    ) -> None:
+        mode = mode.lower()
+        if mode not in self.SUPPORTED_MODES:
+            raise MisconfigurationException(f"`mode` should be either of {self.SUPPORTED_MODES}")
+
+        self._min_lr = min_lr
+        self._max_lr = max_lr
+        self._num_training = num_training
+        self._mode = mode
+        self._early_stop_threshold = early_stop_threshold
+        self._update_attr = update_attr
+
+        self._early_exit = False
+        self.lr_finder: Optional[_LRFinder] = None
+
+    def lr_find(self, trainer: "pl.Trainer", pl_module: "pl.LightningModule") -> None:
+        with isolate_rng():
+            self.optimal_lr = lr_find(
+                trainer,
+                pl_module,
+                min_lr=self._min_lr,
+                max_lr=self._max_lr,
+                num_training=self._num_training,
+                mode=self._mode,
+                early_stop_threshold=self._early_stop_threshold,
+                update_attr=self._update_attr,
+            )
+
+        if self._early_exit:
+            raise _TunerExitException()
```

Comment:
Looks to me like this PR is waiting on #11089 to be merged. As you didn't add the logic to catch this here. Are they meant to be sequenced?

Reply:
yes. check the base :)